src/gallia/log.py

```diff
@@ -258,6 +258,10 @@ def setup_logging(
     logging.logThreads = False
     logging.logProcesses = False
 
+    # Silence python's root logger
+    if len(logger_name) != 0:
```

Comment:
What's the rational of this change? f81561b533cf76f8d2119cc833d92956b2df10ff

Reply:
Since we are not adjusting the root logger, it can be that other python modules that are using the `logging` module produce log output that is not intended, e.g. `numpy` or `matplotlib`.

This change disables these log entries.

Reply:
But then the change should be `if logger_name == "something"`, shouldn't it?

Reply:
No, why should it?

You get the handle of the root logger by calling `logging.getLogger()` without arguments (or with an empty string).
Theoretically, our function `setup_logging` can also be used to configure the root logger directly instead of the default `gallia` logger.

This change is to ensure that the root logger is only silenced, if nobody is attempting to configure exactly the root logger, so basically `if logger_name != ""`, but linters told me checking for empty string is bad behavior.

Reply:
> No, why should it?

It is not clear why this feature is needed.

Further, there is no `numpy` or `matplotlib` in upstream `gallia`.

You mention:

> Theoretically, our function setup_logging can also be used to configure the root logger directly instead of the default gallia logger.

It is not clever to do special things, such as disabling the root logger in a function called `setup_logging()`. People relying on log messages from the root logger might spend hours of debugging and searching this hidden feature.

If I understand this correctly, the suggested code breaks the following code such that no messages are logged:

``` python
import logging

from gallia.log import setup_logging, get_logger

setup_logging()
logger = get_logger("")
logger.info("foo")

logging.info("foo")

logger = get_logger("root")
logger.info("foo")
```

Filtering log messages (by default) from extension modules is out of scope for upstream gallia.

> but linters told me checking for empty string is bad behavior.

This should be fine for linters:

``` python
if variable:
    foo()
```

Reply:
Well, what can I say apart from: "I did not come up with this as a programming exercise, but because we had unsolicited log messages".

If you have such a strong "wont fix" mindset here, I am fine with dropping f81561b533cf76f8d2119cc833d92956b2df10ff alltogether and just live with a few log messages that only exist on CLI. Let's not waste our time...